clvm-traits/src/conversions.rs

```diff
@@ -0,0 +1,104 @@
+use clvmr::{
+    allocator::{NodePtr, SExp},
+    sha2::{Digest, Sha256},
+    Allocator,
+};
+
+use crate::{ClvmValue, FromClvm, FromClvmError, ToClvm, ToClvmError};
+
+/// A wrapper trait for `ToClvm<NodePtr>` that implements conversion to a `NodePtr`.
+pub trait ToPtr: ToClvm<NodePtr> {
+    /// Allocates the value on a given CLVM allocator, returning the corresponding `NodePtr`.
+    fn to_ptr(&self, a: &mut Allocator) -> Result<NodePtr, ToClvmError>;
+}
+
+impl<T> ToPtr for T
+where
+    T: ToClvm<NodePtr>,
+{
+    fn to_ptr(&self, a: &mut Allocator) -> Result<NodePtr, ToClvmError> {
+        self.to_clvm(&mut |value| match value {
+            ClvmValue::Atom(bytes) => match bytes {
+                [] => Ok(a.null()),
+                [1] => Ok(a.one()),
+                _ => Ok(a.new_atom(bytes).or(Err(ToClvmError::LimitReached))?),
+            },
+            ClvmValue::Pair(first, rest) => {
+                Ok(a.new_pair(first, rest).or(Err(ToClvmError::LimitReached))?)
```

Comment:
Maybe I misunderstand something, how come you don't need to call `to_ptr()` on `first` and `rest`? Have they already been converted to allocations in `a`?